openpype/hosts/houdini/plugins/create/create_pointcache.py

```diff
@@ -37,14 +36,30 @@ def create(self, subset_name, instance_data, pre_create_data):
         }
 
         if self.selected_nodes:
-            parms["sop_path"] = self.selected_nodes[0].path()
+            parent = self.selected_nodes[0]
+            parms["sop_path"] = parent.path()
+
+            child_render = ""
+            child_output = ""
 
             # try to find output node
-            for child in self.selected_nodes[0].children():
+            for child in parent.children():
+                if child.isGenericFlagSet(hou.nodeFlag.Render):
+                    child_render = child
                 if child.type().name() == "output":
-                    parms["sop_path"] = child.path()
+                    child_output = child
                     break
 
+            # create output node if not exists
+            if not child_output:
+                child_output = parent.createNode("output", "OUTPUT")
+                child_output.setFirstInput(child_render)
+
+                child_output.setDisplayFlag(1)
+                child_output.setRenderFlag(1)
+                child_output.moveToGoodPosition()
+
+            parms["sop_path"] = child_output.path()
```

Comment:
I still think this is an unexpected change - I wouldn't expect it to create a node, but just to pick the node I had selected.

Reply:
maybe we can add another validator with repair action just for this output node.

Reply:
Still wouldn't know what it would "auto-set" too then.